Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertAnalyticalSurface.cs

```diff
@@ -260,6 +261,172 @@ private Element2D AnalyticalSurfaceToSpeckle(AnalyticalModelSurface revitSurface
 
       return speckleElement2D;
     }
+#else
+    private Element2D AnalyticalSurfaceToSpeckle(AnalyticalPanel revitSurface)
+    {
+
+
+      var speckleElement2D = new Element2D();
+
+      var structuralElement = revitSurface.Document.GetElement(AnalyticalToPhysicalAssociationManager.GetAnalyticalToPhysicalAssociationManager(Doc).GetAssociatedElementId(revitSurface.Id));
+
+
+
+      var mark = GetParamValue<string>(structuralElement, BuiltInParameter.ALL_MODEL_MARK);
+      speckleElement2D.name = mark;
+
+      var edgeNodes = new List<Node> { };
+      var loops = revitSurface.GetOuterContour();
+
+      var displayLine = new Polycurve();
+      foreach (var loop in loops)
+      {
+        var coor = new List<double>();
+
+        var points = loop.Tessellate();
+
+        foreach (var p in points.Skip(1))
+        {
+          var vertex = PointToSpeckle(p);
+          var edgeNode = new Node(vertex, null, null, null);
+          edgeNodes.Add(edgeNode);
+        }
+
+        displayLine.segments.Add(CurveToSpeckle(loop));
+
+      }
+
+      speckleElement2D.topology = edgeNodes;
+      speckleElement2D["displayValue"] = displayLine;
+
+
+      //speckleElement2D.voids = voidNodes;
```

Comment:
I could not find a way to get the voids or inner loops

Reply:
Seems like they've separated the modelling completely. There's a new element called openings. These are voids in Revit2023. Probably need to use these methods instead. Not sure if they have a parent element though to attach to for these openings.